spec/std/file_spec.cr

```diff
@@ -224,6 +224,21 @@ describe "File" do
     end
   end
 
+  describe "open with symnofollow" do
+    it "doesn't follow symlinks" do
+      file = datapath("test_file.txt")
+      other = datapath("test_file.ini")
+
+      with_tempfile("test_file_symlink.txt") do |symlink|
+        File.symlink(File.real_path(file), symlink)
+        File.new(symlink, File::Mode.flags(Read))
```

Comment:
Could we have this test using the `:read` syntax just to make sure it works?

Reply:
Small issue with splat args and extra params.  Is this a crystal bug or intended design?  Do I need to specify types for every parameter?

```
File.open(path, :write, :truncate, perm) do |file|
```

```
Error: no overload matches 'File.open' with types String, Symbol, Symbol, Int32

Overloads are:
 - File.open(filename : Path | String, mode = "r", perm = DEFAULT_CREATE_PERMISSIONS, encoding = nil, invalid = nil)
 - File.open(filename : Path | String, mode : Mode = File::Mode.flags(Read), perm : File::Permissions | Int = DEFAULT_CREATE_PERMISSIONS, encoding = nil, invalid = nil, &block)
 - File.open(filename : Path | String, mode : String, perm = DEFAULT_CREATE_PERMISSIONS, encoding = nil, invalid = nil, &block)
 - File.open(filename : Path | String, *modes : Mode, perm : File::Permissions | Int = DEFAULT_CREATE_PERMISSIONS, encoding = nil, invalid = nil, &block)
```

Reply:
IF I don't specify a type for `mode` It passes the second `Symbol` as `encoding`.

Reply:
Should I keep tests that generate deprecated warnings?  Like `File.new(..., "w")`?

Reply:
Can you fix this by adding more type restrictions to the deprecated modes?

Reply:
Should `File.new(..., "w")` be tested?  If so it generates a warning because it's deprecated.

Reply:
@bcardiff what is the testing strategy for deprecated methods? Should the spec suite be run with warnings off?

Reply:
@bcardiff ?

Reply:
Is there anything I need to do?

Reply:
Let the CI emit the warnings for now. This will push the story to allow warnings in specific call-sites or remind us to finally remove the deprecated methods.

Reply:
Sorry, I have some commits and changes locally for this PR which I haven't published.

Reply:
Bump.  You missed our 6 month anniversary (on this issue).  I'm breaking up with you.

Reply:
I'm so sorry! I forget about things too often >_<